### PR TITLE
Fix performance issue at token transfers tab on Token's page

### PR DESCRIPTION
**Pull request by feliperenan**
Fixes https://github.com/poanetwork/blockscout/issues/1053

### Enhancements
* Denormalized the `block_number` from the transaction to token transfers to avoid joins with the transactions' table and block's table.
* Paginates the results using the `block_number` and the `log_index`. It was using the `inserted_at` that would make the pagination inconsistent.

### Bug Fixes
* The application will be able to open tokens that have several transfers.

## Upgrading
* This need run the migration to add the `block_number` column at token transfer's table. You may need to run the script that copies the `block_number` from the transactions' table to token transfers' table

### Benchmark

* Token tested: `0xd2f81cd7a20d60c0d558496c7169a20968389b40`
* Old query: 6s
* New query: 400ms

Also, sometimes the page didn't open when the users' click on the next page using the `old query`. It works normally with the `new query` introduced at this PR.

### CODE REVIEW

File: apps/explorer/lib/explorer/chain/token_transfer.ex [hunk 1]
```diff
@@ -67,6 +69,7 @@ defmodule Explorer.Chain.TokenTransfer do
     field(:amount, :decimal)
     field(:log_index, :integer, primary_key: true)
     field(:token_id, :decimal)
+    field(:block_number, :integer)
```
Comment (w-sanches) [inline apps/explorer/lib/explorer/chain/token_transfer.ex]: Could we move this guy up to keep alphabetical order? 😶

File: apps/explorer/priv/repo/migrations/scripts/token_transfers_update_block_number_in_batches.sql [hunk 2]
```diff
@@ -0,0 +1,54 @@
+DO $$
+DECLARE
+   row_count integer := 1;
+   batch_size  integer := 100000; -- HOW MANY ITEMS WILL BE UPDATED AT TIME
```
Comment (GuSilverFlame) [inline apps/explorer/priv/repo/migrations/scripts/token_transfers_update_block_number_in_batches.sql]: the batch size seems bigger for this one, does it work ok with a bigger batch?

**Comment by feliperenan** [comment]
I still have to fix the pagination to use the `block_number` and `log_index`. I moved the label back to `wip`.

**Comment by feliperenan** [comment]
@w-sanches @amandasposito and @GuSilverFlame is ready for review again. I've included the pagination changes required.

File: apps/explorer/priv/repo/migrations/scripts/token_transfers_update_block_number_in_batches.sql [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+   iterator  integer := batch_size;
+   affected integer;
+BEGIN
+  DROP TABLE IF EXISTS transactions_temp;
+
+  -- CREATES TEMP TABLE TO STORE DATA TO BE UPDATED
+  CREATE TEMP TABLE transactions_temp(hash bytea, block_number integer, row_number integer);
+
+  INSERT INTO transactions_temp
```
Comment (amandasposito) [inline apps/explorer/priv/repo/migrations/scripts/token_transfers_update_block_number_in_batches.sql]: We may need to change this insert, it is taking too long.

**Pull request merged**